main.tf

```diff
@@ -262,3 +266,44 @@ data "ibm_database_connection" "database_connection" {
   user_id       = ibm_database.elasticsearch.adminuser
   user_type     = "database"
 }
+
+##############################################################################
+# ELSER support
+##############################################################################
+
+locals {
+  es_admin_users = var.enable_elser_model && var.service_credential_names != null && length(var.service_credential_names) > 0 ? [for k, v in var.service_credential_names : k if v == "Administrator"] : []
+  es_admin_user  = length(local.es_admin_users) > 0 ? local.es_admin_users[0] : null
+  es_username    = local.es_admin_user != null ? local.service_credentials_object["credentials"][local.es_admin_user]["username"] : var.admin_pass != null ? "admin" : null
+  es_password    = local.es_admin_user != null ? local.service_credentials_object["credentials"][local.es_admin_user]["password"] : var.admin_pass != null ? ibm_database.elasticsearch.adminpassword : null
+  es_url         = local.es_username != null && local.es_password != null ? "https://${local.es_username}:${local.es_password}@${data.ibm_database_connection.database_connection.https[0].hosts[0].hostname}:${data.ibm_database_connection.database_connection.https[0].hosts[0].port}" : null
+}
+
+resource "null_resource" "put_vectordb_model" {
+  count = var.enable_elser_model ? 1 : 0
+  provisioner "local-exec" {
+    command     = "${path.module}/scripts/put_vectordb_model.sh"
+    interpreter = ["/bin/bash", "-c"]
+    environment = {
+      ES = local.es_url
+    }
+  }
+}
+
+resource "time_sleep" "wait_for_put_trained_model" {
+  count           = var.enable_elser_model ? 1 : 0
+  depends_on      = [null_resource.put_vectordb_model]
+  create_duration = "600s"
```

Comment:
why do we have to wait for 600 seconds? (10 mins is a long time to wait!). Wondering if we can do something more efficiently (like poll for whatever it is we need to wait on)

Reply:
[from the doc](https://cloud.ibm.com/docs/databases-for-elasticsearch?topic=databases-for-elasticsearch-elser-embeddings-elasticsearch)
> You may have to wait a few minutes for the model to finish installing before the start command can execute.

the other way could be, have less minutes added for wait, and try to start the model (by calling the start API), where we have curl command with retry

Reply:
Why don't we just add retry attempts every 30 or 60 seconds in the `scripts/start_vectordb_model.sh` script with an eventual timeout of 10mins? Thats more efficient than just always waiting 10mins